File: BeeCrowds/Array Hash - 1257/Array Hash - 1257 - C/ArrayHash.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>



int main() {
	int caseTest, numStrings, hash;
	char expression[51];
	
	scanf("%d",&caseTest);
	for(int i = 0;i<caseTest;i++) {
		hash = 0;
		scanf("%d\n", &numStrings);
		for(int j = 0;j< numStrings;j++) {
			scanf("%s\n",&expression);

			for(int k = 0;k< strlen(expression);k++) {
				hash += (expression[k] - 'A') + j + k;

	}
		}

	printf("%d\n", hash);
	}

	return 0;

}

File: BeeCrowds/Array Hash - 1257/Array Hash - 1257 - Java/ArrayHash.java
import java.util.*;


public class ArrayHash {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);

		int caseTest, numStrings, hash;

		caseTest = scanner.nextInt();

		for(int i =0;i<caseTest;i++) {
			
			numStrings = scanner.nextInt();                //Utiliza-se a princípio, a abordagem de força bruta, utilizando de 3 laços aninhados, a fim de garantir a leitura dos parâmetros
                                                
			 hash = 0;

			for(int j = 0;j<numStrings;j++) {           //Aproveita-se das variáveis declaradas dentro do laço, para contabilizar a ordem de entrada de cada string

				String expression =  scanner.next();

				for(int k =0;k<expression.length();k++) {                //Assim como para contabilizar a posição do carectere na string.

					hash += (expression.charAt(k) - 'A') + j + k;        //Ao executar (k) - 'A' se retorna a posição do caractere no alfabeto, devido à subtração de valores ASCII.

				}

			}

			System.out.println(hash);

		}

		scanner.close();

	}
}


